### PR TITLE
EOL node engine 6.10 of lambda function

### PR DESCRIPTION
**Pull request by chaspy**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes ~#0000~ Seems that the issue has not been reported

I faced the following error: (refer: https://www.terraform.io/docs/providers/aws/r/lambda_permission.html#example-usage)

```
   │ Error: Error creating Lambda function: InvalidParameterValueException: The runtime parameter of nodejs6.10 is no longer supported for creating or updating AWS Lambda functions. We re
       │ commend you use the new runtime (nodejs10.x) while creating or updating functions.
   │ /Users/take/go/src/github.com/hashicorp/terraform/states/statemgr/filesystem.go:349: [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
   │     status code: 400, request id: 26c1b11c-e092-45d3-bb9f-18d685d26727
```

Unfortunately, nodejs6.10 was EOL on April 30, 2019
ref: https://aws.amazon.com/blogs/developer/node-js-6-is-approaching-end-of-life-upgrade-your-aws-lambda-functions-to-the-node-js-10-lts/

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

None of FEATURES, ENHANCEMENT and BUG FIX. This is just updating doc and test.

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSLambdaFunction_updateRuntime'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSLambdaFunction_updateRuntime -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSLambdaFunction_updateRuntime
=== PAUSE TestAccAWSLambdaFunction_updateRuntime
=== CONT  TestAccAWSLambdaFunction_updateRuntime
--- PASS: TestAccAWSLambdaFunction_updateRuntime (87.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	87.465s
```

### CODE REVIEW

File: aws/resource_aws_lambda_function_test.go [hunk 1]
```diff
@@ -1724,7 +1724,7 @@ resource "aws_lambda_function" "lambda_function_test" {
     function_name = "%s"
     role = "${aws_iam_role.iam_for_lambda.arn}"
     handler = "exports.example"
-    runtime = "nodejs6.10"
+    runtime = "nodejs8.10"
```
Comment (bflad) [inline aws/resource_aws_lambda_function_test.go]: The `TestAccAWSLambdaFunction_updateRuntime` test that references this test configuration should be performing an update of the `runtime` argument and this change will make the runtime value the same as the first test step. Can you please update this to another runtime value (e.g. `nodejs10.x`) and ensure the test check matches the new value? Thanks.

File: aws/resource_aws_lambda_layer_version_test.go [hunk 2]
```diff
@@ -345,7 +345,7 @@ resource "aws_lambda_layer_version" "lambda_layer_test" {
   filename   = "test-fixtures/lambdatest.zip"
   layer_name = "%s"
 
-  compatible_runtimes = ["nodejs8.10", "nodejs6.10"]
```
Comment (bflad) [inline aws/resource_aws_lambda_layer_version_test.go]: The `TestAccAWSLambdaLayerVersion_compatibleRuntimes` test that references this test configuration is currently expecting two values for the `compatible_runtimes` argument:

https://github.com/terraform-providers/terraform-provider-aws/blob/feacc804717a491202318f1ab217f5c76fa049e1/aws/resource_aws_lambda_layer_version_test.go#L168

Can you please add back a second runtime, e.g. `nodejs10.x`

**Comment by chaspy** [comment]
Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSLambdaFunction_updateRuntime'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSLambdaFunction_updateRuntime -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSLambdaFunction_updateRuntime
=== PAUSE TestAccAWSLambdaFunction_updateRuntime
=== CONT  TestAccAWSLambdaFunction_updateRuntime
--- PASS: TestAccAWSLambdaFunction_updateRuntime (88.75s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	88.816s
```

File: aws/resource_aws_lambda_function_test.go (hunk 1 above)
Comment (chaspy) [reply aws/resource_aws_lambda_function_test.go]: Addressed: https://github.com/terraform-providers/terraform-provider-aws/pull/9341/commits/ccfd9575770d6aca8e255dea534d25c924f9d179

File: aws/resource_aws_lambda_layer_version_test.go (hunk 2 above)
Comment (chaspy) [reply aws/resource_aws_lambda_layer_version_test.go]: Addressed: https://github.com/terraform-providers/terraform-provider-aws/pull/9341/commits/95128f1568196332ae8fd16297269bb6be4c6bc8

**Comment by chaspy** [comment]
@bflad thank you for your review! Ready for review again 👍

**Pull request merged**